Improve Workspace Docs: Add Deletion Info, Fix Typos, and Update Alerts (#510)
layer5io/docs: content/en/cloud/spaces/workspaces.md

You can create Workspaces to group related Designs and Environments, define domains of responsibility for your teams, and streamline resource management.

## Key Features

- **Resource Sharing:** Workspaces allow for seamless resource sharing among team members, fostering collaboration.
- **Logical Grouping:** Within Workspaces, you can group related components such as environments and infrastructure designs.
- **Flexibility:** Workspaces support various resources like Kubernetes, Prometheus, Jaeger, Nginx, and more.
- **Simplified Management:** Managing and deploying resources is made easy within Workspaces.
- **Access Control:** Workspaces allow you to control access to resources by granting permissions to users and teams.

### Workspace Relationships and Restrictions

- Access to Workspaces may be granted to one or more teams.
- As a point of collaboration to facilitate work, Workspaces may have zero or more Environments associated.
- A Workspace is closely associated with Teams in Meshery. Teams are groups of users with varying permissions, and they are at the center of resource access and management within a Workspace.

> - One or more teams can be assigned to a workspace.

Reviewer: ```suggestion
- One or more teams can be assigned access to a workspace.
```

> - Same team can be assigned to multiple workspaces.
> - A team can be assigned to a workspace only if the team is not already assigned to another workspace.

Reviewer: ```suggestion
```

> After creating a Workspace, one of your next steps is to resource that Workspace. Like a shared drive (or shared collection of files). Workspaces are your Google Drive, while Meshery Designs are your Google Docs.

Reviewer: ```suggestion
After creating a Workspace, one of your next steps is to assign team access, create or transfer designs into the Workspace, and allocate resources that to Workspace by associating it with one or more Environments, therein making the Environment's connections available to users of the Workspace. 

Like a shared drive (or shared collection of files), Workspaces are your Google Drive, while Meshery Designs are your Google Docs.
```

### Workspace Deletion Behavior

When a Workspace is deleted:

- Deletion is **permanent and irreversible**. Proceed with caution.

> - Associated mappings between the Workspace and its Environments, Designs, Teams, and Views are detached.

Reviewer: ```suggestion
- Any associated Teams or Environments with the Workspace are detached. 
```

- A Workspace is closely associated with Teams in Meshery. Teams are groups of users with varying permissions, and they are at the center of resource access and management within a Workspace.
- One or more teams can be assigned to a workspace.
- Same team can be assigned to multiple workspaces.
- A team can be assigned to a workspace only if the team is not already assigned to another workspace.

After creating a Workspace, one of your next steps is to resource that Workspace. Like a shared drive (or shared collection of files). Workspaces are your Google Drive, while Meshery Designs are your Google Docs.

### Workspace Deletion Behavior

When a Workspace is deleted, Meshery follows a non-destructive deletion strategy to preserve important data:

- The Workspace itself is **soft-deleted** by setting a `deleted_at` timestamp. It is no longer active but still exists in the database for auditing purposes.
- **Associated mappings** between the Workspace and its Environments, Designs, Teams, and Views are also soft-deleted.  
- **Environments, Designs, Teams, and Views themselves are not deleted**. They remain available and can be reassigned to other Workspaces if needed.

> - Users will see the previously associated resources listed as available again, ready for new associations.

Reviewer: I'm confused by this statement. Can you offer an example?
